lib/player.js

```diff
@@ -5717,6 +5717,8 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
     const loaded = this.stats_ && this.bufferObserver_ && this.playhead_;
 
     if (loaded) {
+      const defaultRate = this.playRateController_.getDefaultRate() || 1.0;
+      this.playRateController_.set(isBuffering? 0: defaultRate);
```

Comment:
This would appear to revert part of #5696, but not the whole change.  We shouldn't merge this without a bigger discussion in the conversation tab of this PR.

Reply:
Agree with you and that the reason i was sharing the concern about reintroducing the isBuffering_ value in the playbackrate controller.
Happy to update the PR and keep the Unit testing that is checking all of that.

Reply:
To do this behavior I only see a three possibilities...
- Remove autoplay, and use play in this case
- Use pause/play
- Use plackback rate
For me the best option is the first...

Reply:
I'm a bit puzzled here. removing auto play it's not against the video player behavior ?
I will expect to setup Shaka player in auto play mode and still respecting the reBufferingGoal or the minbuffer of the dash.

Reply:
the goal is to respect the buffer duration and for that we have no other solution rather than stopping the playback.

1. removing the Auto play look wired to me. but happy to better understand your point.
2. Play, pause will trigger wired event at the playback start and during the buffering.
3. use playback rate was quite a nice way without too much impact but mybe i'm wrong.

Reply:
>     1. removing the Auto play look wired to me. but happy to better understand your point.

My point is, if before starting playback we know that we are in this case, if the video element has the autoplay attribute, we delete it, wait for the initial buffering to finish, and then we manually play and reset the attribute. With this we prevent unwanted play and pause events from being launched.

Reply:
hum, I'm just thinking about the use case of the buffering we will have during the playback. 
During the playback, we can have buffering that will happened. in that use case the auto play can't be a solution because we already played. 
additionally, the configuration can be changed by the integrator as soon as you have buffering state over the playback.
in that use case we should still continue to respect the rebuffering or minbuffertime.
I'm mainly working on this use case for my app.

during the playback

- activating the play rate management
- if buffer fill velocity reduce and rate reduce
- we reduce the rebuffering goal up to 3 chunks
- if buffering event is trigger
- we increase the rebuffering goal to ensure that the buffer ahead is enough
- this reduce the number of rebuffering state
- re update the expected configuration when playback restart properly


Sorry for the long comment, i'm not saying it's a worng idea for to manage the autopay but not sure it's managing all the states

Reply:
maybe the bufferPoller_ should not exist because we have not proper event management observer.
Also more unit test should be done to ensure that we have not regression.

Reply:
@Iragne Well, the goal is to reverse the changes of https://github.com/shaka-project/shaka-player/pull/5696 and improve it with what you mention. Once you have made the changes, ping me and I will review it! Thank you!

Reply:
ok will do @avelad

Reply:
@avelad  just push the update

Reply:
I have only made two comments, the rest is perfect.